### DISCUSSION TITLE
AACencoder  vs SBCencoder encoder.begin difference

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#740] [Q&A] [answered]
i was gong through this old example of AACencoder , i notice one thing that you didnot feed sampling bits or sampling frequency to AACencoder , while to SBC encoder we must feed these data to encoder . why is this difference ? 

Also their is no information what is the sampling rate and bits of a2dp data ,below just read data stream  and write to encoder file , 

// Get the audio data from A2DP to save it in a file
void read_data_stream(const uint8_t *data, uint32_t length) {
    if (encoder!=nullptr) encoder.write(data, len);
}
 

https://github.com/pschatzmann/arduino-audio-tools/blob/ba7653fd670ee279b840293ce270387e17d71379/sandbox/a2dp-file_aac/a2dp-file_aac.ino

**Comment by pschatzmann** [comment, marked as answer]
All encoders need to get the AudioInfo as you can see in the [documentation](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_audio_encoder.html). This is usually done via the [EncodedAudioStream](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_encoded_audio_stream.html) by calling the begin(AudioInfo) method!

This is also nicely explained in the [Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/Encoding-and-Decoding-of-Audio)
